server/application/application.go

```diff
@@ -1569,8 +1569,13 @@ func (s *Server) PodLogs(q *application.ApplicationPodLogsQuery, ws application.
 		return nil
 	}
 
-	if len(pods) > maxPodLogsToRender {
-		return errors.New("Max pods to view logs are reached. Please provide more granular query.")
```

Comment:
this is the root error that ends up in the argocd throwing 500 error responses. 

while I agree having a configurable limit may mitigate this to some extent, can we also solve the generic case and treat this error differently?

a 4xx response may be more fitted.

Reply:
As the specifying the log query will fix the problem I took the `codes.InvalidArgument` which will be transformed to `http.StatusBadRequest`

Reply:
Maybe the variable should be removed here also?
https://github.com/argoproj/argo-cd/blob/b12630c4be3dc73bc89eb5295ef3a582c7ae628b/server/application/application.go#L68

Reply:
Thanks @Jellyfrog for a good catch! I removed unused const.